BrainPortal/app/views/users/show.html.erb

```diff
@@ -172,13 +172,16 @@
                :class  => "button",
                :method => :post,
                :data   => { :confirm => "Are you sure you want to unlink your account with this Globus identity?" }
-             )
-         %>
-       <% else %>
-         (No Globus identity linked to your account)<br>
-         <%= link_to 'Link a Globus identity', @globus_uri, :class => 'button globus_button' %>
-       <% end %>
-    <% end %>
+              )
```

Comment:
I don't understand the view logic here.

If the viewing the page as an admin user, we still get the unlink button, which if pressed will not work properly.

Why not just take the view code as it was originally and JUST put a `if current_user.id == @user.id` around the buttons, and nothing else?

Reply:
The unlink button was not mentioned in the issue.

Reply:
I know it wasn't and it's good your thought about it, but the fix is wrong. Just hide it.

Reply:
sure, putting condition around each button. Then the Globus section cell itself will remain, albeit potentially empty.

Reply:
It won't be totally empty, see above, it would say `Provider name:` and `Provider user:`

Reply:
Just look at the original code... if you put the if around each button, it works for both normal users and admin users.